### PR TITLE
Fix masking registration

### PR DESCRIPTION
**Pull request by hjmjohnson**

### CODE REVIEW

**Comment by hjmjohnson** [comment]
@reginakim Please review this patch set.  Thank you for investigating.

File: AutoWorkup/utilities/configFileParser.py [hunk 1]
```diff
@@ -34,6 +34,15 @@
...
 from . import misc
 
+# http://stackoverflow.com/questions/715417/converting-from-a-string-to-boolean-in-python
+def str2bool(v):
+  if str(v).lower() in ("yes", "true", "t", "1"):
+      return True
+  elif str(v).lower() in ("no", "false", "f", "0"):
+      return False
+  print( "ERROR: INVALID String to bool conversion for '{0}'".format(v) )
+  raise
```
Comment (hjmjohnson) [inline AutoWorkup/utilities/configFileParser.py]: Codacy Issue found: [The raise statement is not inside an except clause](https://www.codacy.com/app/hans-j-johnson/BRAINSTools/file/2704433953/issues/source?bid=3194574&fileBranchId=3342661#l44)

File: AutoWorkup/workflows/WorkupT1T2TissueClassify.py [hunk 2]
```diff
@@ -137,21 +138,16 @@ def CreateTissueClassifyWorkflow(WFname, master_config, InterpolationMode,UseReg
...
-
         tissueClassifyWF.connect(fixedROIAuto, 'outputROIMaskVolume',A2SantsRegistrationPreABCRigid,'fixed_image_mask')
-        tissueClassifyWF.connect(movingROIAuto, 'outputROIMaskVolume',A2SantsRegistrationPreABCRigid,'moving_image_mask')
-
         tissueClassifyWF.connect(fixedROIAuto, 'outputROIMaskVolume',A2SantsRegistrationPreABCSyN,'fixed_image_mask')
-        tissueClassifyWF.connect(movingROIAuto, 'outputROIMaskVolume',A2SantsRegistrationPreABCSyN,'moving_image_mask')
+
+    ## NOTE: Always use atlas head region to avoid computing this every time.
+    tissueClassifyWF.connect(inputsSpec, 'atlasheadregion',A2SantsRegistrationPreABCRigid,'moving_image_mask')
+    tissueClassifyWF.connect(inputsSpec, 'atlasheadregion',A2SantsRegistrationPreABCSyN,'moving_image_mask')
```
Comment (reginakim) [inline AutoWorkup/workflows/WorkupT1T2TissueClassify.py]: @hjmjohnson 
Did you put this intentionally here, outside of the 'if UseRegistrationMasking == True'? What would happen if the statement is false, and the only moving_image_mask is given to the registration? 

Regina

Comment (hjmjohnson) [inline AutoWorkup/workflows/WorkupT1T2TissueClassify.py]: Is it possible for ANTs to take only the moving image?    If so, then yes that would be what was intended.  For the atlas, we always have a good mask, so we should use it.  For the fixed image, it is not always clear if we can generate a good mask.

**Pull request merged**